### PR TITLE
More powerful patch script

### PR DESCRIPTION
**Pull request by poeschlr**
The patch script was quite basic.

This now allows to simply specify the config directory and the scheme directory and everything will be updated at once (with flags for disabling specific definitions)
It also uses the argparse class which means we now have documentation via the --help option.
And error handling.

---

Edit: A possible future improvement would be the option to run it in reverse to allow it to create definition files.

### CODE REVIEW

**Comment by pointhi** [comment]
please update the README as well

**Comment by poeschlr** [comment]
Readme is updated. The script now also backs up the settings file before a change is made.

File: patch.py [hunk 1]
```diff
@@ -1,20 +1,121 @@
 #!/bin/python3.6
```
Comment (pointhi) [inline patch.py]: not your fault, but this is a bit to specific

Proper way:

```python
#!/usr/bin/env python3
```

Comment (poeschlr) [reply patch.py]: Damn i noticed it at the beginning and then forgot to fix it. Done now.

**Comment by pointhi** [comment]
thanks

**Pull request merged**